test/extended/operators/olm.go

```diff
@@ -105,4 +105,19 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
 		}
 	})
 
+	// OCP-21130 - [bug ALM-736]Fetching non-existent `PackageManifest` does not return 404
+	// author: bandrade@redhat.com
+	g.It("Fetching non-existent `PackageManifest` does not return 404", func() {
+		msg, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("packagemanifest", "--all-namespaces", "--no-headers").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		packageserverLines := strings.Split(msg, "\n")
+		if len(packageserverLines) > 0 {
+			raw, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("packagemanifest", "a_package_that_not_exists", "-o yaml", "--loglevel=8").Output()
+			o.Expect(err).To(o.HaveOccurred())
+			o.Expect(raw).To(o.ContainSubstring("\"code\": 404"))
```

Comment:
Sorry, this is confusing. Is this testing checking that the package server does or doesn't return a 404 when asking for a non existent package? This test seems like it passes when there is a 404, but the test name, pr header, and commit message all state the opposite. Should we update the test name to `Fetching non-existent PackageManifest returns a 404`?

Reply:
I'm sorry, the bug was written in this form because 404 was not returned before it was fixed. And this test case was written to avoid regression using the bug name. I'll correct to avoid this confusion.

Reply:
Changed as requested